lib/internal/streams/pipeline.js

```diff
@@ -181,6 +182,8 @@ function pipeline(...streams) {
     const reading = i < streams.length - 1;
     const writing = i > 0;
 
+    finish.bind({ index: i });
```

Comment:
Note: [Function.prototype.bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) creates a new function, so that line has no effect as is. You would need to do something like:

```js
const finishWithIndex = finish.bind({ index: i });

// ... and below, change occurrences of `finish` in the function scope
destroys.push(destroyer(ret, false, true, finishWithIndex));
```

Not a very elegant variable name, I'm very bad at naming those things 😅